staging_vespalib/src/vespa/vespalib/util/singleexecutor.cpp

```diff
@@ -56,11 +63,27 @@ SingleExecutor::setTaskLimit(uint32_t taskLimit) {
     _wantedTaskLimit = vespalib::roundUp2inN(taskLimit);
 }
 
+void
+SingleExecutor::drain(Lock & lock) {
+    uint64_t wp = _wp.load(std::memory_order_relaxed);
+    while (numTasks() > 0) {
+        _consumerCondition.notify_one();
+        sleepProducer(lock, 100us, wp);
```

Comment:
letting go of the lock here will allow multiple threads to call drain at the same time. this might be ok, but could lead to creating more new buffers than you really need.

Reply:
I think that is not a normal usecase.